### PR TITLE
only refresh invalid lines

### PR DESCRIPTION
**Pull request by jixiuf**

### CODE REVIEW

**Comment by VanLaser** [comment]
Assuming there are no problems with this code and it gets integrated, this means only the mouse handling remains to be added? These latest emacs-libvterm pulls really seem to be moving the project to a beta/usable stage :) :+1:

File: vterm.el [hunk 1]
```diff
@@ -187,10 +197,22 @@ be send to the terminal."
...
+
+
+(defun vterm--run-timer (buffer)
+  (interactive)
+  (let ((inhibit-redisplay t)
+        (inhibit-read-only t))
+    (when (buffer-live-p buffer)
+      (with-current-buffer buffer
+        (vterm--update vterm--term)))))
+
```
Comment (akermu) [inline vterm.el]: Is there any particular reason to change the event-driven I/O to timer-based polling?

Comment (jixiuf) [reply vterm.el]: I think don't need to response to  all I/O  event .

Comment (TheBB) [reply vterm.el]: My terminals are by and large idle 99% of the time. I think the event-driven updates were fine, and nothing about that design prevents us from ignoring updates on certain events if that is ever a problem.

Comment (jixiuf) [reply vterm.el]: ok the timer is removed

**Comment by akermu** [comment]
tmp

**Pull request merged**